cmd/collector/app/server/http_test.go

```diff
@@ -58,3 +85,176 @@ func TestSpanCollectorHTTP(t *testing.T) {
 	assert.NoError(t, err)
 	assert.NotNil(t, response)
 }
+
+func TestSpanCollectorHTTPS(t *testing.T) {
+
+	testCases := []struct {
+		name              string
+		TLS               tlscfg.Options
+		clientTLS         tlscfg.Options
+		expectError       bool
+		expectClientError bool
+		expectServerFail  bool
+	}{
+		{
+			name: "should fail with TLS client to untrusted TLS server",
+			TLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				ServerName: "example.com",
+			},
+			expectError:       true,
+			expectClientError: true,
+			expectServerFail:  false,
+		},
+		{
+			name: "should fail with TLS client to trusted TLS server with incorrect hostname",
+			TLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "nonEmpty",
+			},
+			expectError:       true,
+			expectClientError: true,
+			expectServerFail:  false,
+		},
+		{
+			name: "should pass with TLS client to trusted TLS server with correct hostname",
+			TLS: tlscfg.Options{
+				Enabled:  true,
+				CertPath: testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:  testCertKeyLocation + "/example-server-key.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "example.com",
+			},
+			expectError:       false,
+			expectClientError: false,
+			expectServerFail:  false,
+		},
+		{
+			name: "should fail with TLS client without cert to trusted TLS server requiring cert",
+			TLS: tlscfg.Options{
+				Enabled:      true,
+				CertPath:     testCertKeyLocation + "/example-server-cert.pem",
+				KeyPath:      testCertKeyLocation + "/example-server-key.pem",
+				ClientCAPath: testCertKeyLocation + "/example-CA-cert.pem",
+			},
+			clientTLS: tlscfg.Options{
+				Enabled:    true,
+				CAPath:     testCertKeyLocation + "/example-CA-cert.pem",
+				ServerName: "example.com",
+			},
+			expectError:       false,
+			expectServerFail:  false,
+			expectClientError: true,
+		},
+		{
+			name: "should pass with TLS client with cert to trusted TLS server requiring cert",
+			TLS: tlscfg.Options{
+				Enabled:      true,
+				CertPath:     testCertKeyLocation + "/example-server-cert.pem",
```

Comment:
This is an example of mTLS right? @davidcmitchell  @yurishkuro

Reply:
Yes. Once this PR lands, the client mTLS will be _technically_ feasible, but most Jaeger SDKs do not actually support attaching client certificates to requests (Java SDK might).

Also, mTLS seems like a lot of management overhead. If TLS from client is really necessary, I would consider using plain TLS in combination with plain auth in the headers, which I think Jaeger clients already support (but Jaeger backed ignores it afaik).

Reply:
I would push for mTLS since it's frequently a security requirement for interservice communication. Otherwise, should we create an issue to implement plain auth alongside TLS (so that it's not ignored by the jaeger backend)?

Reply:
Feel free to open a discussion on that. The challenge with supporting plain auth is how you manage the allowed credentials - do we need to invent some special config file listing all user names & passwords? Do we need to integrate with some credentials store like LDAP?